openstack/identity/v3/tokens/results.go

```diff
@@ -41,6 +41,36 @@ type ServiceCatalog struct {
 	Entries []CatalogEntry `json:"catalog"`
 }
 
+// Domain provides information about the domain to which this token grants access.
+type Domain struct {
+	ID   string `json:"id"`
+	Name string `json:"name"`
+}
+
+// User represents a user resource that exists on the API.
+type User struct {
+	Domain Domain `json:"domain"`
+	ID     string `json:"id"`
+	Name   string `json:"name"`
+}
+
+// Role provides information about roles to which User is authorized.
+type Role struct {
+	ID   string `json:"id"`
+	Name string `json:"name"`
+}
+
+type Roles struct {
+	Roles []*Role `json:"roles"`
```

Comment:
> Slices are already pointers

👍 

Thanks for catching the other ones, too. I'm going to see if there's a `vet` or lint gate test that can catch those.